utils/s2n_safety.c

```diff
@@ -171,3 +171,11 @@ int s2n_mul_overflow(uint32_t a, uint32_t b, uint32_t* out)
     *out = (uint32_t) result;
     return S2N_SUCCESS;
 }
+
+int s2n_add_overflow(uint32_t a, uint32_t b, uint32_t* out)
```

Comment:
You need a unit test for this.  You can steal the one I wrote for aws-c-common https://github.com/awslabs/aws-c-common/blob/master/tests/math_test.c